tools/roxvet/common/traverse.go

```diff
@@ -0,0 +1,62 @@
+package common
+
+import (
+	"go/ast"
+
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+// FilteredPreorder calls `inspector.Preorder(nodeTypes, fn)`, but filters out all files that do not pass the given
+// fileFilter.
+func FilteredPreorder(inspector *inspector.Inspector, fileFilter FileFilter, nodeTypes []ast.Node, fn func(n ast.Node)) {
+	effTypes := nodeTypes[:len(nodeTypes):len(nodeTypes)]
+	hadFile := hasFile(nodeTypes)
+	if !hadFile {
+		effTypes = append(effTypes, (*ast.File)(nil))
+	}
+	skip := false
```

Comment:
I won't block this PR, but I've been told `var skip bool` is preferred, so I usually tell others this, too. However, this isn't a core part of the product, so I don't mind as much

Reply:
Good point, will change this

Reply:
```suggestion
	var skip bool
```